src/device-registry/utils/get-measurements.js

```diff
@@ -44,13 +47,27 @@ const getMeasurements = async (
           callbackErrors(err, req, res);
         } else {
           const filter = generateEventsFilter(startTime, endTime, device);
+
+          let deviceDetail = await getDetail(tenant);
+          logElement("number of devices", deviceDetail.length);
+
           let skipInt = skip ? skip : 0;
-          let limitInt = limit ? limit : 50;
-          let events = await getModelByTenant(
+          let limitInt = limit ? limit : deviceDetail.length;
+
+          let allEvents = await getModelByTenant(
             tenant,
             "event",
             EventSchema
           ).list({ skipInt, limitInt, filter });
+
+          let recentEvents = await getModelByTenant(
```

Comment:
utilising the static method for getting all the recent events for each device in the network